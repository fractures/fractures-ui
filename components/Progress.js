import React, { Component, Fragment } from "react"

const defaultProps = {
	max: 100,
	value: 0
}

class Progress extends Component {
	render() {
		return (
			<Fragment>
				<progress className="fr-progress radius-max w-100" max={ this.props.max } value={ this.props.value } />
				<style jsx global>{`
					.fr-progress {
						height: 0.5rem;

						background-color: var(--shade-1);
						border: 0;
					}

					.fr-progress::-webkit-progress-bar {
						background-color: transparent;
					}

					.fr-progress::-webkit-progress-value {
						background-color: var(--shade-4);
						border-radius: 1rem;
					}
				`}</style>
			</Fragment>
		)
	}
}

Progress.defaultProps = defaultProps

export default Progress
