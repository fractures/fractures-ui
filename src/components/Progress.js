import React, { Component, Fragment } from "react"

const defaultProps = {
	max: 100,
	value: 0
}

class Progress extends Component {
	render() {
		return (
			<Fragment>
				<progress className="fr-progress radius-max w-100" max={ this.props.max } value={ this.props.value } />
				<style jsx global>{`
					.fr-progress {
						height: 0.5rem;

						background-color: var(--fr-100);
						border: 0;
					}

					.fr-progress::-webkit-progress-bar {
						background-color: transparent;
					}

					.fr-progress::-webkit-progress-value {
						background-color: var(--fr-500);
						border-radius: 1rem;
					}
				`}</style>
			</Fragment>
		)
	}
}

Progress.defaultProps = defaultProps

export default Progress
