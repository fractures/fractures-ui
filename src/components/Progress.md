### Progress

```js noeditor
<Fractures theme="blue" />
```

```js
<Progress value={25} />
```
