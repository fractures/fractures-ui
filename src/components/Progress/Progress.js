import PropTypes from "prop-types"
import React from "react"
import styled from "styled-components"

const propTypes = {
	className: PropTypes.string,
	max: PropTypes.number,
	value: PropTypes.number.isRequired
}

const defaultProps = {
	max: 100,
	value: 0
}

const NakedProgress = props => <progress className={ props.className } max={ props.max } value={ props.value } />

const Progress = styled(NakedProgress)`
	height: 0.5rem;
	width: 100%;

	background-color: var(--fr-100);
	border: 0;
	border-radius: 10rem;

	&::-webkit-progress-bar {
		background-color: transparent;
	}

	&::-webkit-progress-value {
		background-color: var(--fr-500);
		border-radius: 1rem;
	}
`

NakedProgress.propTypes = propTypes
NakedProgress.defaultProps = defaultProps

Progress.propTypes = propTypes
Progress.defaultProps = defaultProps

export default Progress
