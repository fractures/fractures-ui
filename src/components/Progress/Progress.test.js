import Progress from "./Progress.js"

describe("Progress", () => {
	const componentRender = <Progress value={ 31 } />
	const component = renderer.create(componentRender)
	const tree = component.toJSON()

	it("Renders correctly", () => {
		expect(tree).toMatchSnapshot()
	})
})
