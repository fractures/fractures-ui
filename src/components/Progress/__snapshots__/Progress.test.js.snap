// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress Renders correctly 1`] = `
<progress
  className="sc-bdVaJa czqDfu"
  max={100}
  value={31}
/>
`;
